source/common/config/subscription_factory_impl.cc

```diff
@@ -103,9 +108,8 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
           std::move(mux), callbacks, resource_decoder, stats, type_url, dispatcher_,
           Utility::configSourceInitialFetchTimeout(config), /*is_aggregated*/ false, options);
     }
-    default:
-      NOT_REACHED_GCOVR_EXCL_LINE;
     }
+    PANIC_DUE_TO_CORRUPT_ENUM;
```

Comment:
Out of curiosity, since you added additional explicit and implicit coverage above of the various cases, is this still needed or are all cases now covered?

Reply:
gcc requires handling in case the enum is not in range so it's still necessary

Reply:
OK, so the clang runs will ensure we have full coverage of all cases and the gcc ones will still need this. It might make sense to actually annotate this as _FOR_GCC or the like. Otherwise LGTM.